### PR TITLE
Optimise initial sync speed

### PR DESCRIPTION
**Pull request by aguycalled**
This PR optimises the synchronisation speed from the genesis block.

### CODE REVIEW

**Comment by mxaddict** [comment]
Noice! Will test in a few hours!

File: src/main.cpp [hunk 1]
```diff
@@ -5228,6 +5225,7 @@ static bool ActivateBestChainStep(CValidationState& state, const CChainParams& c
 
         // Connect new blocks.
         for(CBlockIndex *pindexConnect: boost::adaptors::reverse(vpindexToConnect)) {
+            int64_t nTime1 = GetTimeMicros();
```
Comment (mxaddict) [inline src/main.cpp:5228]: Is this needed?

**Comment by mxaddict** [comment]
Judging from the changes, I think the new added script checkpoints will speed up sync alot.

**Comment by chasingkirkjufell** [comment]
Waiting for the gitian build to test

**Comment by navbuilder** [comment]
A new build of 84aa5132f3c4a3923ff8b6b2263f76d777ab6296 has completed succesfully!
Binaries available at https://build.nav.community/binaries/optimize-initial-sync-speed

**Comment by mxaddict** [comment]
I'm running a full sync with the gitian build on windows 10.

will report back the speed once done.

**Review by mxaddict — APPROVED** [review]
This binary loaded 50% of the chain in first 22 hours, while master build would take about 4 days to sync full chain.

**Comment by aguycalled** [comment]
last commits optimize sync even further

**Comment by mxaddict** [comment]
> last commits optimize sync even further

New changes look good, will test gitian build once ready

**Comment by navbuilder** [comment]
A new build of 754fbe543b4c4c89b794cec432e607e4381f00f1 has completed succesfully!
Binaries available at https://build.nav.community/binaries/optimize-initial-sync-speed

**Comment by navbuilder** [comment]
A new build of 08a53ec67bb911dafca666ef1edc793a1e7c30b2 has completed succesfully!
Binaries available at https://build.nav.community/binaries/optimize-initial-sync-speed

**Comment by mxaddict** [comment]
@aguycalled  build 08a53ec worked really nice, I got a full sync in less than 24 hours, about 22 hours from genesis to fully synced.

**Comment by mxaddict** [comment]
> @aguycalled build [08a53ec](https://github.com/navcoin/navcoin-core/commit/08a53ec67bb911dafca666ef1edc793a1e7c30b2) worked really nice, I got a full sync in less than 24 hours, about 22 hours from genesis to fully synced.

Was getting about 4-5% per hour sync.

**Comment by navbuilder** [comment]
A new build of 68c5a9a745411d1770c94628182c09db1d90ee89 has completed succesfully!
Binaries available at https://build.nav.community/binaries/optimize-initial-sync-speed

**Comment by mxaddict** [comment]
> A new build of [68c5a9a](https://github.com/navcoin/navcoin-core/commit/68c5a9a745411d1770c94628182c09db1d90ee89) has completed succesfully!
> Binaries available at https://build.nav.community/binaries/optimize-initial-sync-speed

Testing this new binary with the bootstrap.

**Comment by mxaddict** [comment]
> > A new build of [68c5a9a](https://github.com/navcoin/navcoin-core/commit/68c5a9a745411d1770c94628182c09db1d90ee89) has completed succesfully!
> > Binaries available at https://build.nav.community/binaries/optimize-initial-sync-speed
> 
> Testing this new binary with the bootstrap.

Sync from bootstrap no longer getting stuck

**Review by mxaddict — APPROVED** [review]

**Comment by chasingkirkjufell** [comment]
syncing from scratch at the moment.

**Review by chasingkirkjufell — APPROVED** [review]
tested gitian build on windows 10

**Pull request merged**